aws/resource_aws_route53_resolver_endpoint.go

```diff
@@ -303,7 +303,11 @@ func route53ResolverEndpointRefresh(conn *route53resolver.Route53Resolver, epId
 			return nil, "", err
 		}
 
-		return resp.ResolverEndpoint, aws.StringValue(resp.ResolverEndpoint.Status), nil
+		status := aws.StringValue(resp.ResolverEndpoint.Status)
+		if status == route53resolver.ResolverEndpointStatusActionNeeded {
+			return nil, status, errors.New(aws.StringValue(resp.ResolverEndpoint.StatusMessage))
```

Comment:
General rule is to not trust values upstream if an error is returned.  Is there a reason to return status along with the error? 

Should  we provide context to this error or is StatusMessage clear enough so that the the end user knows how to act on it?

Reply:
Good point. Setting the error means that `status` is ignored in `WaitForState`.
Cleanest may be to return `nil` error and `status` and log the `StatusMessage`?

Reply:
I used the same logic as in `aws_ram_principal_association`: https://github.com/terraform-providers/terraform-provider-aws/blob/190da406cb71111424b7b91f99dc1bccf5873c62/aws/resource_aws_ram_principal_association.go#L186-L189.